### PR TITLE
docs: update Setfit ABSA tutorial

### LINKED ISSUE #4695: [DOCS] Tutorial on SetFit ABSA
## Which page or section is this issue related to?

Add a tutorial showcasing https://huggingface.co/docs/setfit/en/how_to/absa using Argilla.

### PR DESCRIPTION
**Pull request by sdiazlor**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #4695 

**Type of change**

(Remember to title the PR according to the type of change)

- [x] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes.)

- [x] `sphinx-autobuild` (read [Developer Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation) for more details)

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Review by davidberenstein1957 — COMMENTED** [review]
Looking great, some nits.

Why do we need to SpanQuestions?
Why do we use two models here?
```python
model = AbsaModel.from_pretrained(
    "sentence-transformers/paraphrase-mpnet-base-v2",
    "sentence-transformers/paraphrase-MiniLM-L6-v2",
    spacy_model="en_core_web_lg",
)
```
I feel you might need to explain `ordinal` a bit.
Perhaps we can create more catching title? Just spitballing so feel free to improve. "Enhancing sentiment classification with span based polarity predictions using SetFit" Not everyone will know aspact based sentiment analysis because it is not very descriptive IMO.

File: docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb [hunk 1]
```diff
@@ -0,0 +1,1079 @@
...
+    }
+   },
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "Sentiment analysis is an effective natural language processing approach for assessing the emotional tone of text. However, measuring the overall sentiment can be challenging. **Aspect-Based Sentiment Analysis (ABSA)** refines this task by classifying sentiments toward different detected aspects within the text. For instance, consider the sentence \"The food was great, but the service was terrible\". In this case, ABSA identifies the positive sentiment toward \"food\" and the negative sentiment toward \"service. This nuanced approach provides a detailed understanding of the sentiment, which is invaluable for capturing consumers' more subtle opinions about specific products or services.\n",
```
Comment (ignacioct) [inline docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb]:
```suggestion
    "Sentiment analysis is an effective natural language processing approach for assessing the emotional tone of text. However, measuring the overall sentiment can be challenging. **Aspect-Based Sentiment Analysis (ABSA)** refines this task by classifying sentiments toward different detected aspects within the text. For instance, consider the sentence \"The food was great, but the service was terrible\". In this case, ABSA identifies the positive sentiment toward \"food\" and the negative sentiment towards \"service"\. This nuanced approach provides a detailed understanding of the sentiment, which is invaluable for capturing consumers' more subtle opinions about specific products or services.\n",
```

File: docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb [hunk 2]
```diff
@@ -0,0 +1,1079 @@
...
+    "\n",
+    "Normally, ABSA can be divided into three main steps:\n",
+    "\n",
+    "1. **Aspect Extraction**: This stage involves identifying potential aspects or entities mentioned in the text. In the sentence \"The food was great, but the service was terrible\", the aspects are \"food\" and \"service\".\n",
+    "2. **Aspect Classification**: This involves verifying the accuracy of the identified aspects. That is, if \"food\" and \"service\" are indeed the aspects that need to be analyzed.\n",
+    "3. **Polarity Classification**: This step involves assessing the sentiment associated with each identified aspect. What is the sentiment toward \"food\" and \"service\"?\n",
```
Comment (ignacioct) [inline docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb:36]:
```suggestion
    "3. **Polarity Classification**: This step involves assessing the sentiment associated with each identified aspect. What is the sentiment towards \"food\" and \"service\"?\n",
```

File: docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb [hunk 3]
```diff
@@ -0,0 +1,1079 @@
...
+    "train_df = train_eval_split['train'].to_pandas()\n",
+    "eval_df = train_eval_split['test'].to_pandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, it's time to format the dataset for Setfit ABSA. The datasets should have four columns: `text`, `span`, `label`, and `ordinal`. The `text` column contains the text to be analyzed, `span` is the aspect, `label` refers to the sentiment of the aspect, and `ordinal` indicates the index of the span occurrence if appears multiple times (e.g. if \"food\" appears 3 times in a review, but it is only identified as aspect when mentioned the 2nd time, the ordinal would be 1). So, what about if a review has multiple aspects? In this case, we will duplicate the review for each aspect. For instance,\n",
+    "\n",
```
Comment (ignacioct) [inline docs/_source/tutorials_and_integrations/tutorials/feedback/training-setfit-absa.ipynb]: "what about if ..." sounds a little bit off in my head. I'd maybe change it to "what happens if ..."

**Review by ignacioct — CHANGES_REQUESTED** [review]
Code looks good to me, and I found a few typos but nothing serious. Good job :)

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**